### ISSUE TITLE
Garden of Qu'aph: tanks can punt people from the upper deck into the arena

### ISSUE THREAD

**Issue by RoR-Zomega** [WarEmu/WarBugs#17171]
**Expected behavior and actual behavior:**

The invisible wall around the edge of the upperdeck to the arena should be high enough that nothing can pass over it. However tanks can use their knockback to send people over even if those people are not on higher ground. Confirmed with a KOTBS with the knock-back tactic.

**Steps to reproduce the problem:**

1) Be a KOTBS
2) Equip knockback tactic
3) Find a destro character on the upper deck near-ish the edge
4) Use Repel on them

The destro player will be punted into the arena.

**Testing Screenshots/Videos/Evidences (always needed):**

Sorry, didn't grab a SS of this one.

**Comment by dalen** [comment]
this is possible, but the player punted down will get teleported up again when they hit the ground

**Comment by RoR-Zomega** [comment]
True, it's just a griefing tool. The person who got knocked down has a moment or two to do some damage or throw out some heals before getting teleported back up though.

**Comment by dalen** [comment]
The ability to damage players dueling has been fixed. Will fix ability to heal or buff players dueling as well soon.

If we have higher invisible walls it would stop players from being able to target people dueling, and then you can't really follow fights as you can't see debuffs etc.

**Comment by RoR-Zomega** [comment]
Apologies if this gets too technical - I'm not sure who knows what about the NIF file format.

I went digging as I know there are areas in the game that block movement but not targeting and I managed to find one. The asset "em_city_ship_galleon02.nif" has invisible walls that block movement (you can't jump off the boat into the sea) but they don't block targeting.

Using NifSkope I checked out the names and the properties of the different nodes. In all cases I found (including other nif files that had invisible walls you could click through) I found:

- If the collision geometry blocked movement and targeting then it was under a node named "collidee"
- If the collision geometry blocked movement only then it was under a node named "nopick"

I managed to track down the NIF file for the Gardens (much fun seeing as I don't have a full hashed filenames list... in the end I went for the biggest file size that I had no name for and it worked!) and noticed that all of the collision geometry is under "collidee" nodes, including the invisible walls overlooking the arena.

In theory - and I do really mean in theory - if Yali is able to split the walls overlooking the arena into their own mesh (currently they are merged with the floor and outer wall triangles) and then when building the Nif file get them into a node named "nopick" then you should be able to raise the height as much as you want and still be able to target people below.

I checked everything else I could find from the data shown in Nifskope (materials, properties, flags, etc) and for invisible colliding geometry these are all the same regardless of whether you can target through or not. The only different I could find was the click-through stuff being under a node named "nopick". Even if this turns out to be wrong, examining that ship model in more detail might give you the answers to making invisible walls you can click through.

Crap-load of screenshots incoming.

**Comment by RoR-Zomega** [comment]
This "collidee" geometry blockes movementg and targeting. It is the north boat on Altdorf docks and you can confirm by trying to click NPCs through the ropes going from the ship to the top of the mast.

![ship_collidee](https://user-images.githubusercontent.com/60266177/85201905-c121b000-b2fa-11ea-93a8-1fe601cf9c41.PNG)

These 2 screenshots show the "nopick" geometry. The "nopick" node is under a "collidee" note but this doesn't seem to matter. I haven't found any files where "collidee" was a child of "nopick" but I did find some where "collidee" and "nopick" were siblings. I'm assuming the game uses whichever name is further down the tree, i.e. in this example the "nopick" child node takes priority over the "collidee" parent node.

![ship_nopick1](https://user-images.githubusercontent.com/60266177/85202039-9d129e80-b2fb-11ea-84dc-74871ec6493d.PNG)
![ship_nopick2](https://user-images.githubusercontent.com/60266177/85202041-9d129e80-b2fb-11ea-93fa-e00d96a32a9b.PNG)

**Comment by RoR-Zomega** [comment]
Garden map showing walls overlooking arena are currently merged with some floor collision geometry and under a "collidee" node.

![garden_collidee](https://user-images.githubusercontent.com/60266177/85202152-45286780-b2fc-11ea-8409-d3497e01636c.PNG)

**Comment by RoR-Zomega** [comment]
Update: so I was a naughty boy and just amended the node name from "collidee" to "nopick" in Nifskope for the Garden map and then reinserted the file back into art2.myp. Game ran fine but I couldn't click through the walls.

Further examination of the galleon Nif file is needed to work out exactly why those 2 sets of geometry don't block targeting.

**Comment by RoR-Zomega** [comment]
I can confirm changing the name of the node to "nopick" will tell the client to ignore it when targeting with the mouse. Screenshow showing the modified version of the Garden with updated node names. I'm able to mouse-over target things visible right above the floor of the upper deck which usually would be impossible. My movement is still blocked as before but I can click though.

![Zomega_445](https://user-images.githubusercontent.com/60266177/85209424-b59cac00-b32f-11ea-9f13-7600ed9c90c5.jpg)

Updated model in Nifskope showing the renamed node. Please not that this fix isn't suitable for the live server as it allows you to click through all of the collision geometry selected (i.e. including most of the floors). Yali or someone will need to recompile the map with the collision walls overlooking the arena as their own mesh. Those specific walls can then be set to "nopick" to let you click through them without affecting the floor/etc.

![image](https://user-images.githubusercontent.com/60266177/85209457-1cba6080-b330-11ea-80db-868bac7c6ff7.png)
